app/Main.hs

```diff
@@ -62,6 +64,18 @@ hsOutP =
     optional $ strOption $
     long "hs-out" <> metavar "DIR" <> help "Generate Haskell bindings to specified dir"
 
+elmVersionP :: Parser (Maybe ElmVersion)
+elmVersionP =
+    mkElmVersion <$>
+    (optional $ strOption $
```

Comment:
You can use the `eitherReader` from optparse-applicative to directly show errors. It would also be good to list the choices in the help text.